pkg/controller/kogitodataindex/kogitodataindex_controller.go

```diff
@@ -84,6 +91,8 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	watchOwnedObjects := []runtime.Object{
 		&corev1.Service{},
 		&appsv1.StatefulSet{},
+		&routev1.Route{},
+		&kafkabetav1.KafkaTopic{},
```

Comment:
@xiezhang7 since you're adding the `KafkaTopic` to the watch list, you should also catch the error and display a WARN message during the watch adding routine (line 103). Look this excerpt for an example:
https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/controller/kogitoinfra/kogitoinfra_controller.go#L75-L78

@sutaakar this is related to the problem you reported. Many thanks for looking into it.

Reply:
This will only happen when the Kafka CRD is not installed in the cluster (e.g. missing Strimzi).

Reply:
Thanks @sutaakar @ricardozanini . It is fixed: https://github.com/kiegroup/kogito-cloud-operator/blob/1ca5a3b3ed9c77f331405bd7aa556d1a34d3bb79/pkg/controller/kogitodataindex/kogitodataindex_controller.go#L105